js/preload/preloadMain.js

```diff
@@ -229,6 +232,36 @@ function createAPI() {
             }
         },
 
+        /**
+         * Allows JS to open a file picker window
+         * @param options {Electron.OpenDialogOptions}
+         * @param callback
+         * @example
+         * openFilePicker({
+         *  type: 'openFile',
+         *  title: 'test',
+         *  multiSelect: true,
+         *  filters: filters: [{ name: 'test', extensions: ['JPG', 'TXT', 'ai'] }]
+         * })
+         */
+        openFilePicker: function (options, callback) {
+            if (options && typeof callback === 'function') {
+                const browserWindow = remote.getCurrentWindow();
+                const { type, title, defaultPath, multiSelect, filters } = options;
+                const properties = [ type ];
+                if (multiSelect) properties.push('multiSelections');
+                remote.dialog.showOpenDialog(browserWindow, { title, defaultPath, properties, filters }, (filePaths) => {
+                    if (!filePaths) return;
+                    const files = filePaths.map((file) => {
+                        const mimeType = mime.lookup(file);
+                        const filename = file.substring(file.lastIndexOf('/') + 1);
+                        return new File([ fs.readFileSync(file) ], filename, { type: mimeType });
```

Comment:
If the file(s) is big & the system is slow, the UI thread will be blocked. Please use `async` version.